### DISCUSSION TITLE
Meta: Legends of Lighthning Tournament Plan

### DISCUSSION THREAD

**Discussion by bonomat** [itchysats/10101#4] [General]
The [tournament Legends of Lightning](https://makers.bolt.fun/tournaments/1/overview) runs officially from 12th Oct - 8th Dec 2022 which gives us 8 weeks to hack, or 8 iterations. 
Iteration 1 is over by 19.10.2022.

As a reference, our project was announced here: https://makers.bolt.fun/project/10101.

# Project Goal

We are building **10101**: the one-stop app for Bitcoiners. 
A trading app with built in Lightning support for trading and investing. Features include (but are not limited to): derivatives, Taro for spot trading and DCA for user-onboarding. We bring Defi back to Bitcoin and to the masses! 

To summarize our goal for this tournament: we will

1. Build a mobile wallet
2. Integrate the existing protocol
3. Lift everything on ⚡️
4. Integrate more protocols
5. Go 🥜

The idea of this discussion is to have an overview of what is planned to achieve our goal and to describe how we to get there. 

From our experience, the hardest part is always the remaining 20%, we should keep that in mind when planning the iterations. 

# Iterations (subject to change): 

1. Planning: https://github.com/itchysats/10101/discussions/4#discussioncomment-3874660
2. Kick-off: Project setup! Goal: Mobile wallet
3. Hacking: Integrate existing protocol! Goal: build a _light client_ going against an ItchySats daemon using the mobile wallet
4. Hacking: Go Lightning! Goal: a PoC to have LN and ItchySats in one protocol
5. Hacking: ...
6. Hacking: ...
7. Wrapping up: ...
8. Preparing for presentation: ...

**Comment by bonomat** [comment]
# Iteration 1: 12.10.2022 - 18.10.2022 

## Goal 

Come up with a strategy! It should not be a waterfall-style project plan but we need a clear vision on what to present at the end of the tournament. A breakdown into the 8 iterations will help us to keep an overview on what we need to achieve by when to get to reach our goal.

## Outcome

A strategy (see this discussion)

**Reply by holzeis** [reply]
I propose the following scope for the legend of lightning tournament.

- Move the signing keys from the ItchySats taker daemon to the TenTenOne Mobile Wallet.

## Protocol Adaptions
The following sequence diagram illustrate on a high level the required adaptations on the cfd trading flow to integrate a mobile wallet.

![tentenonemobilewalletcfdtrading](https://user-images.githubusercontent.com/382048/195866715-7dddf0e2-75cd-4098-915c-e67b81706fac.png)

As the keys will remain with Satoshi, we could run the taker as well managed - non-custodial, but not self sovereign. However, if we go that path, we need to make the taker multi-tenant able, which it isn't today.

**Reply by holzeis** [reply]
I created an issue to summarize the required tasks - https://github.com/itchysats/itchysats/issues/3189

**Reply by da-kami** [reply]
Here's what is in my head. Might be a bit crazy to fit this into 8 weeks, but I think if we go a bit nuts we could potentially achieve this. Given the we already have some days we team members are off I am not sure we can get this into 8 weeks. But I thought I'll give it a try either way. We can always downscope :)

Given that we pack quite a lot in short time we have to be aware that this will be fast, i.e. we should plan to throw away and re-write things afterwards. The fast-track path that I outline below should help us shape our vision and try things, it should *not* be the code-foundation of everything we build on top later. Given that we are not very good at throwing things away we have to evaluate if this path forward is suitable for us.

One unknown / hard to decide: Current production. 
    - If we want to fix the monitoring "at the side" this will take time away that we don't have.
    - What if there is unforeseen issues?

---

Overall goal: An app that *outlines* the vision of 10101 and integrates ItchySats.

I would propose that we work towards `demo` each week.

1. Refine plan for (2. and 3.)
2. Mob session on the mobile user interface (2 days). Once we have a rough wireframe we can parallelize mocking different parts of the application. We can still work closely together here. I roughly imaging:
	0. Login (would not focus on that, add later)
	1. Dashboard: Wallet overview, access to "trading / investment services" 
	2. Wallet details
	3. Onboarding services
	4. Trading services
		1. ItchySats as service
3. Split the team into 2, one that works on POC on Lightning integration and one that continues with the app. Goals:
	- Refine the mocks to get into a stage where we start implementing. Most of the implementation will be "fake", but ItchySats should be usable within the app. (Can be connected to e.g. Umbrel setup, no full phone setup)
	- Lightning POC to be defined
4. Goals
	- Have a first version of the app that is usable (without ItchySats)
	- Lightning POC scope to be defined
5. Goals
	- Integrate ItchySats into the app "somehow" 
	- Lightning POC scope to be defined
6. See if there is any chance of getting Lightning POC together with the app team
7. 🤯
8. Presentation prep, goal: transport the vision of 10101
    - the app as prototype that shows UI
    - the Lightning POC that shows the vision of "we can do this on Lightning"

**Reply by bonomat** [reply]
@da-kami : I like the idea of having the frontend first and align the team on what the outcome should look like so that we always have the end goal in mind.

What do you think of not going for a mob-session on the frontend though but splitting earlier and having a sub-team building the LN PoC while the others are building the app, i.e. from your points, move the LN PoC parts of point 3. into point 2.

**Reply by da-kami** [reply]
That also works - mob sessions are hard to control and the outcome is more unknown. Happy to see an earlier split, but then it might be harder to align on what the vision of 10101 in "everybody's head" in terms of a user interface. We might get a better big picture if we do this together - but it's hard to predict if it actually helps.

Instead of mobbing (🤪) we can also do fast feedback cycles where the frontend team presents to the LN team on a frequent (e.g. daily) basis.

**Reply by bonomat** [reply]
> I propose the following scope for the legend of lightning tournament.
> 
>     * Move the signing keys from the ItchySats taker daemon to the TenTenOne Mobile Wallet.
> 

Do you propose to do this in iteration 1 or as the overall scope? I think for the whole tournament we need a bit more. Something along the lines what @da-kami proposed. It's aggressive, ambitious and doable 💪

**Reply by bonomat** [reply]
> Instead of mobbing (🤪) we can also do fast feedback cycles where the frontend team presents to the LN team on a frequent (e.g. daily) basis.

That's a good idea.

**Reply by holzeis** [reply]
> > I propose the following scope for the legend of lightning tournament.
> > ```
> > * Move the signing keys from the ItchySats taker daemon to the TenTenOne Mobile Wallet.
> > ```
> 
> Do you propose to do this in iteration 1 or as the overall scope? I think for the whole tournament we need a bit more. Something along the lines what @da-kami proposed. It's aggressive, ambitious and doable 💪

well it was the proposal for the overall scope - It depends on how many resources we want to dedicate to that endeavor. I tried to split the LN integration into another discussion (see below).

**Reply by holzeis** [reply]
The way I read the tournament announcement is that we have time until the 24th of November. 

> 12th Oct - Project entries open
24th Nov - Project entries close
28th Nov - Judging panels announce track finalists
6th Dec - Finalists pitch their projects
7th Dec - Track winners + grand champions announced.
https://makers.bolt.fun/tournaments/1/overview. 

That will leave us with 6 iterations (with the first iteration almost over already) until the projects will be judged. The projects will be judged by the following categories.

> 1). 🎯 Value Proposition: Does the project have a product market fit? Does it provide value to the bitcoin ecosystem and beyond?
> 2). 🚨 Innovation: Is it something we've seen before or does it bring something new and exciting to bitcoin and beyond?
> 3). ⚡️ Bitcoin Integration & Scalability: Have they used bitcoin/lightning? If so, how many features? How well will this product scale for either local or global adoption?
> 4). 👁️ Transparency (#BuildInPublic): Has the project’s team been transparent throughout their product design and development journey?
> 5). ✅ Execution: How well has the project been executed?
> 6). 🍒 UI/UX Design: How well has the application or feature been designed?
> 7). 🔥 Je ne sais quoi: Does the project have that extra level of pizazz or coolness? Does it raise the bar


## Transparency!
We need to write a weekly PPP (Plans, Progress, Problems) report - https://makers.bolt.fun/story/ppps-week-1--160 as well as post a milestones update here and then. The first PPP should already be written for the current iteration.

## Innovation & Value Proposition
We should write up a post on https://makers.bolt.fun/project/10101 on itchysats. I think we can reuse quite a lot here.

## Bitcoin Integration & Scalability
We should write up a post on https://makers.bolt.fun/project/10101 on itchysats. and engage the community for feedback.

## Execution

I suggest the following approach.

### 1. Iteration (12.10 - 19.10.): 
Decide on the cupcake scope and outline the birthday cake scope. Instead of splitting up the team and working on multiple things in parallel - I think its safer if we pick one milestone, build that and then go over to the next one. I am not saying we have to spent all the time on the initial mobile wallet, but experience shows that the devil lives in the details and I'd rather have a little bit done, than a lot almost done.

- Published 1. PPP

### 2. Iteration (20.10-26.10): (Philipp off: 3 days, Richie off: 2 days)
 - Setup mobile wallet project. (decide on a framework)
 - Decide on a wallet framework (porting the bdk wallet using neon, use a typescript wallet implementation, etc.)
 - Build wireframes (UX Design) for the overall user experience and publish a post describing flow on https://makers.bolt.fun/project/10101
 - Published 2. PPP

### 3. Iteration (27.10-02.11): (Daniel off: 2 days)
Reiterate over the wireframes and start polishing them. Would be great if we've got feedback on our project post.

- Publish polished UI/UX design - Initialise wallet using Bip39.
- Wallet view.
- Design background tasks and push notifications
- Publish 3. PPP

### 4. Iteration (03.11-09.11)

- Trades overview. Show offers.
- Place an order using the keys generated on the phone.
- Write a post about transparency & value proposition.
- Publish 4. PPP

### 5. Iteration (10.11-16.11)

- Rollover notification and triggering from the phone.
- Collborative settlement triggered from the phone.
- Write a post about bitcoin integration & scalability.
- Publish 5. PPP

### 6. Iteration (17.11-24.11)

- Testing
- Cleanup
- Design LN wallet integration and write a post about it.
- Publish 6. PPP


### 25.11-7.12.

If we make it to the finals, we can focus here on the LN integration.

**Comment by holzeis** [comment]
Depending on the resources we are willing to put in this tournament / use case, we could as well start to integrate with a lightning wallet - as has already been discussed through funding the cfd respectively as target wallet for payouts.
